internal/common/splunk/hostid.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunk
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/cloud"
+)
+
+// HostIDKey represents a host identifier.
+type HostIDKey string
+
+const (
+	// AWS
+	HostIDKeyAWS HostIDKey = "AWSUniqueId"
+	// GCP
+	HostIDKeyGCP HostIDKey = "gcp_id"
+)
+
+// HostID is a unique key and value (usually used as a dimension) to uniquely identify a host.
+type HostID struct {
+	// Key is the key name/type.
+	Key HostIDKey
+	// Value is the unique ID.
+	ID string
+}
+
+// resourceAttributesToDimensions will return a set of dimension from the
+// resource attributes, including a cloud host id (AWSUniqueId, gcp_id, etc.)
+// if it can be constructed from the provided metadata.
+func ResourceToHostID(res pdata.Resource) (HostID, bool) {
```

Comment:
Is this used anywhere? I cannot find. Is the intent that it will be used in a future PR?

Reply:
Not yet, will be in future PR (APM correlation).